### DISCUSSION TITLE
Decal node should not use the AABB for decal projection as it interferes with the editors snap_to_floor method.

### DISCUSSION THREAD

**Discussion by nan0m** [godotengine/godot-proposals#10301] [3D]
The Decal node, should not use the AABB for projection. I find it hard to explain but consider these two cases. 
(Alternatively we should be able to exclude objects from being considered for a scenes AABB.)

Case 1: 
I want to add a decal around the base of a tree I place in a scene. If I want to project the decal around the base of the tree, the aabb of the decal must exceed that of the tree, for projection to happen. However, from now on, the tree.tscn with the decal won't snap to the floor with the base of the tree, but with the lower bound of the decal node's aabb. I will now have neither the projection of the decal on the ground, nor will the tree be placed on the ground.

Case 2: I want to project an AABB on the floor.  I place it in the scene, I press snap_to_floor. Since the decal is placed on the floor based on it's aabb, there is no projection visible on the floor, nor happening. I have to manually change it's position on the y axis, to have projection work again.

**Comment by Calinou** [comment]
I would modify the `snap_to_floor()` method's behavior instead of changing how Decal renders, as we can't break compatibility with existing projects by changing how decals are placed in all projects retroactively.

By contrast, `snap_to_floor()` is a one-time editor-only operation, so we can change its behavior without breaking compatibility.
